esmvaltool/cmorizers/data/cmorizer.py

```diff
@@ -0,0 +1,486 @@
+
+        # put settings in environment
+        env = dict(os.environ)
+        env['settings'] = settings_file
+        env['esmvaltool_root'] = esmvaltool_root
+        env['cmor_tables'] = str(
+            Path(esmvalcore.cmor.__file__).parent / 'tables')
+        logger.info("Using CMOR tables at %s", env['cmor_tables'])
+        # call NCL
+        ncl_call = ['ncl', script]
+        logger.info("Executing cmd: %s", ' '.join(ncl_call))
+        process = subprocess.Popen(ncl_call,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.STDOUT,
+                                   env=env)
+        output, err = process.communicate()
+        for oline in str(output.decode('utf-8')).split('\n'):
+            logger.info('[NCL] %s', oline)
+        if err:
+            logger.error('[NCL][subprocess.Popen ERROR] %s', err)
+            return False
+        return True
+
+    def _run_pyt_script(self, in_dir, out_dir, dataset, start, end):
+        """Run the Python cmorization mechanism."""
+        module_name = ('esmvaltool.cmorizers.data.formatters.datasets.' +
+                       dataset.lower().replace("-", "_"))
+        module = importlib.import_module(module_name)
+        logger.info("CMORizing dataset %s using Python script %s", dataset,
+                    module.__file__)
+        cmor_cfg = read_cmor_config(dataset)
+        module.cmorization(in_dir, out_dir, cmor_cfg, self.config, start, end)
+        return True
+
+
+class DataCommand():
+    """Download and format data to use with ESMValTool."""
+    def __init__(self):
+        datasets_file = os.path.join(os.path.dirname(__file__), 'datasets.yml')
+        with open(datasets_file) as data:
+            self._info = yaml.safe_load(data)
+        self.formatter = Formatter(self._info)
+
+    def list(self):
+        """List all supported datasets."""
+        print()
+        print(f'| {"Dataset name":30} | Tier | Auto-download | Last access |')
+        print('-' * 71)
+        for dataset, dataset_info in self._info['datasets'].items():
+            date = datetime.datetime.strptime(str(dataset_info['last_access']),
+                                              "%Y-%m-%d")
+            print(f"| {dataset:30} | {dataset_info['tier']:4} "
+                  f"| {self.formatter.has_downloader(dataset):13} "
+                  f"|  {date.strftime('%Y-%m-%d')} |")
+        print('-' * 71)
+
+    def info(self, dataset):
+        """Show detailed info about a specific dataset.
+
+        Parameters
+        ----------
+        dataset : str
+            dataset to show
+        """
+        dataset_info = self._info['datasets'][dataset]
+        print(dataset)
+        print()
+        print(f"Tier: {dataset_info['tier']}")
+        print(f"Automatic download: {self.formatter.has_downloader(dataset)}")
+        print("")
+        print(dataset_info['info'])
```

Comment:
Would it be possible to display the source as well? For datasets for which the automatic download is not supported, this would help to find the data online

Reply:
Done!